instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/utils.py

```diff
@@ -132,6 +132,9 @@ def attach_span(task, task_id, span, is_publish=False):
     NOTE: We cannot test for this well yet, because we do not run a celery worker,
     and cannot run `task.apply_async()`
     """
+    if task is None:
+        return
```

Comment:
so I tested this against the following repo  https://github.com/goatsthatcode/opentelemetry-instrumentation-celery-example which was reproducing the disconnected scan use case.  As far as I can tell it seems to be nesting and closing spans correctly but I could be missing something as well.  I just opened the upstream PR (moved this to my company GitHub since that was the request from my company) here https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1052